modules/ROOT/pages/migration/index.adoc

```diff
@@ -16,14 +16,389 @@ To update your Neo4j GraphQL Library, use npm or the package manager of choice:
 npm update @neo4j/graphql
 ----
 
+== What's new
+
+Until version 5 of the Library, we tried to limit, as much as possible, the number of breaking changes and related migration pain. In the meantime, we introduced new features such as Aggregation, Fulltext search, Vector search, Connection API, and Interfaces&Union support, to name a few! This approach led to an inconsistent user-facing API with room for ambiguous behaviors.
+
+Several moons ago, we introduced support for the GraphQL Cursor Connections Specification https://relay.dev/graphql/connections.htm / https://relay.dev/docs/tutorial/connections-pagination/.
+This was a great addition, not only added the Cursor-based pagination in the Library but also gave us the space to support relationship properties.
+
+However, it created a diverse experience between the Connection API and the normal one, in some cases, even within the same query.
+
+=== New Connection API 
```

Comment:
As discussed, I would avoid giving a name here, this is just the API, not "Connection"